metric-anomaly-task-manager/src/main/java/org/hypertrace/alert/engine/metric/anomaly/task/manager/job/DbRuleSource.java

```diff
@@ -0,0 +1,147 @@
+import org.apache.commons.lang3.tuple.Pair;
+import org.hypertrace.alert.engine.metric.anomaly.datamodel.rule.source.RuleSource;
+import org.hypertrace.config.service.v1.ConfigServiceGrpc;
+import org.hypertrace.config.service.v1.ConfigServiceGrpc.ConfigServiceBlockingStub;
+import org.hypertrace.config.service.v1.ContextSpecificConfig;
+import org.hypertrace.config.service.v1.GetAllConfigsRequest;
+import org.hypertrace.core.documentstore.Document;
+import org.hypertrace.core.documentstore.JSONDocument;
+import org.hypertrace.core.grpcutils.client.RequestContextClientCallCredsProviderFactory;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+import org.hypertrace.core.serviceframework.spi.PlatformServiceLifecycle;
+import org.hypertrace.notification.config.service.v1.GetAllNotificationRulesRequest;
+import org.hypertrace.notification.config.service.v1.NotificationRule;
+import org.hypertrace.notification.config.service.v1.NotificationRuleConfigServiceGrpc;
+
+public class DbRuleSource implements RuleSource {
+
+  private static final String EVENT_CONDITION_DATA_KEY = "eventConditionData";
+  private static final String EVENT_CONDITION_MUTABLE_DATA_KEY = "eventConditionMutableData";
+  private static final String NOTIFICATION_RULE_MUTABLE_DATA_KEY = "notificationRuleMutableData";
+  private static final String METRIC_ANOMALY_DATA_KEY = "metricAnomalyEventCondition";
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+  private final ConfigServiceBlockingStub configServiceBlockingStub;
+  private final NotificationRuleConfigServiceGrpc.NotificationRuleConfigServiceBlockingStub
+      notificationRuleStub;
+  private final String resourceNamespace;
+  private final String resourceName;
+  private final List<String> tenantIds;
+
+  public DbRuleSource(Config dataStoreConfig, PlatformServiceLifecycle lifecycle) {
+    ManagedChannel configChannel =
+        ManagedChannelBuilder.forAddress(
+                dataStoreConfig.getString("config.service.host"),
+                dataStoreConfig.getInt("config.service.port"))
+            .usePlaintext()
+            .build();
+    lifecycle.shutdownComplete().thenRun(configChannel::shutdown);
+    this.resourceName = dataStoreConfig.getString("resourceName");
+    this.resourceNamespace = dataStoreConfig.getString("resourceNamespace");
+    configServiceBlockingStub =
+        ConfigServiceGrpc.newBlockingStub(configChannel)
+            .withCallCredentials(
+                RequestContextClientCallCredsProviderFactory.getClientCallCredsProvider().get());
+    notificationRuleStub =
+        NotificationRuleConfigServiceGrpc.newBlockingStub(configChannel)
+            .withCallCredentials(
+                RequestContextClientCallCredsProviderFactory.getClientCallCredsProvider().get());
+    this.tenantIds = dataStoreConfig.getStringList("tenantIds");
+  }
+
+  @Override
+  public List<Document> getAllRules(Predicate<JsonNode> predicate) {
```

Comment:
`Predicate<JsonNode> predicate` is this used?

Reply:
It is not being used as of now, since the same predicate cannot be used for both json and db based rules. 
I some refactoring is required in this interface, will check that once these changes are in.